lesson17/index.js

```diff
@@ -0,0 +1,171 @@
+const sliderDirections = ["previous", "next"];
+
+const controlSlide = (direction) => {
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+  const showSlide = currentSlide[direction];
+  currentSlide.setAttribute("data-hidden","true");
+  showSlide.setAttribute("data-hidden","false");
+  controlBtnBehavior();
+  updateFractionNum();
+}
+
+const slideImgTemplate = (index, slide) => {
+  const div = document.createElement("div");
+  div.classList.add("sliderImg");
+  div.id = `slide-${index}`;
+
+  index === 0
+    ? div.setAttribute("data-hidden", "false")
+    : div.setAttribute("data-hidden", "true");
+
+  const img = document.createElement("img");
+  img.src = slide.img;
+  div.appendChild(img);
+  return div;
+};
+
+const initSlides = (slides)=> {
+  const slider = document.createElement("div");
+  slider.classList.add("slider");
+  slider.id = "js-slider"
+  const sliderWrapper = document.createElement("div");
+  sliderWrapper.classList.add("sliderWrapper");
+  sliderWrapper.id = "js-sliderWrapper"
+  slider.appendChild(sliderWrapper);
+  main.appendChild(slider);
+
+  const fragment = document.createDocumentFragment();
+  for (const [index, slide] of slides.entries()) {
+    fragment.appendChild(slideImgTemplate(index, slide));
+    sliderWrapper.appendChild(fragment);
+  }
+  createSliderBtns();
+}
+
+const setBtnEvent = () => {
+  const buttons = document.querySelectorAll(".sliderBtn");
+  buttons.forEach((button) => {
+    button.addEventListener("click", (e) => {
+      const eventTargetBtnType = e.target.getAttribute("data-btntype");
+      controlSlide(`${eventTargetBtnType}ElementSibling`);
+    });
+  });
+}
+
+const createSliderBtns = () => {
+  sliderDirections.forEach((sliderDirection) => {
+    const button = document.createElement("div");
+    button.classList.add("sliderBtn", `${sliderDirection}`);
+    button.id = `js-slider-${sliderDirection}Btn`;
+    button.setAttribute("data-btntype", `${sliderDirection}`)
+    sliderDirection === "previous" && button.classList.add("is-disabled");
+    main.querySelector(".slider").appendChild(button);
+  });
+  setBtnEvent();
+}
+
+const controlBtnBehavior = () => {
+  const allSlides = Array.from(document.querySelectorAll(".sliderImg"));
+  const allSlidesLength = Array.from(document.querySelectorAll(".sliderImg")).length;
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+  const prevBtn = document.getElementById("js-slider-previousBtn");
+  const nextBtn = document.getElementById("js-slider-nextBtn");
+  const currentSlideIndex = allSlides.findIndex((slide)=>{
+    return slide === currentSlide;
+  });
+
+  if (currentSlideIndex === 0) {
+    prevBtn.classList.add("is-disabled");
+    return;
+  } else if(currentSlideIndex === (allSlidesLength - 1)) {
+    nextBtn.classList.add("is-disabled");
+    return;
+  } else {
+    prevBtn.classList.remove("is-disabled");
+    nextBtn.classList.remove("is-disabled");
+  }
```

Comment:
else ifにreturnを書くのではなく、処理を複数のif文に分けて書いたほうが可読性の高いコードになると思って前回IMOしました。言葉足らずですみません。もう一度ご確認いただけたら嬉しいです！

[早期リターンを書こう](https://zenn.dev/media_engine/articles/early_return)
```suggestion
  if (currentSlideIndex === 0) {
    prevBtn.classList.add("is-disabled");
    return;
  }
  if (currentSlideIndex === allSlidesLength - 1) {
    nextBtn.classList.add("is-disabled");
    return;
  }
  prevBtn.classList.remove("is-disabled");
  nextBtn.classList.remove("is-disabled");
```

Reply:
[d4dfe5f](https://github.com/hippopo-moo/js-handson-work/pull/19/commits/d4dfe5facc365abe7058d82a5698e2b6d46cdb62)

Sorry, I was completely premature...><!!
It only makes sense if you separate the if statements, not the elseif.
Thanks for pointing that out!
Fixed it!